x/stakeibc/keeper/lsm.go

```diff
@@ -151,13 +151,20 @@ func (k Keeper) GetValidatorFromLSMTokenDenom(denom string, validators []*types.
 	}
 	validatorAddress := split[0]
 
-	// Confirm validator is in Stride's validator set and does not have an active slash query in flight
+	// Confirm the validator:
+	//  1. is in Stride's validator set
+	//  2. does not have an active slash query in flight
+	//  3. has a known exchange rate
 	for _, validator := range validators {
 		if validator.Address == validatorAddress {
 			if validator.SlashQueryInProgress {
 				return types.Validator{}, errorsmod.Wrapf(types.ErrValidatorWasSlashed,
 					"validator %s was slashed, liquid stakes from this validator are temporarily unavailable", validator.Address)
 			}
+			if validator.InternalSharesToTokensRate.IsNil() || validator.InternalSharesToTokensRate.IsZero() {
+				return types.Validator{}, errorsmod.Wrapf(types.ErrValidatorExchangeRateNotKnown,
```

Comment:
So an LSM liquid stake tx from a validator we don't yet have the rate for will fail... but should we also try to kick off an async (ie. the user is not waiting for it) ICQ to see if we can now get the exchange rate?

Reply:
If we don't have the exchange rate yet, that should indicate that there's a query in progress (it's possible if we get a liquid stake the second after a validator was added)

Reply:
Oh do we kick off an ICQ right when a new val is added?

Reply:
And do validators start with nil exchrate or exchrate=1?

Reply:
If we both start them at nil exchrate and kick off an ICQ by default upon adding them, then this logic makes sense. otherwise I don't understand the case where the exchrate would be nil or zero.

Reply:
It’s set to nil by default when the val is added and then the query for the exchange rate is submitted immediately